### ISSUE TITLE
Possible bug: Text Node positions changes after calling getClientRect in translated stage

### ISSUE THREAD

**Issue by borsh** [konvajs/konva#843]
Here is a code to reproduce a bug(may be I misunderstand something). 

```js
var stage = new Konva.Stage({
  container: 'container',   
  width: 400,
  height: 400
});

var layer = new Konva.Layer();

var rect = new Konva.Rect({
  x : 120, 
  y : 120,
  width : 150,
  height : 50,
  stroke : 'black',
  strokeWidth: 4
  
} )

var text00 = new Konva.Text({
  text :'Sample text',
  x :130,
  y : 130,
  fontSize : 20
})

layer.add(rect);
layer.add(text00)


stage.add(layer);

stage.x(+40)
stage.y(+40)

stage.draw();

layer.removeChildren()
text00.getClientRect({}) // Commenting this line or putting it after line 41 puts text in rectangle

layer.add(text00)
layer.add(rect)


stage.draw();
```

Expected behavior : Text inside rectangle
Reproduced behavior: Text outside rectangle

Commenting line with  text00.getClientRect({}) puts text back inside the rectangle

Here is a link to codepen

[https://codepen.io/kborsh/pen/dyoyZaG](https://codepen.io/kborsh/pen/dyoyZaG)

**Comment by friksa** [comment]
I have seen similar issues and found a workaround by not reusing Shapes or Transforms.  Once you remove them from the layer, create new versions of them and you will be fine.

**Comment by lavrton** [comment]
Thanks for the good demo. The bug is fixed. Published in `4.1.4`
